### PR TITLE
Verify reindexing is run after index mode change

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs and @geirst please review.

### CODE REVIEW

**Comment by jonmv** [comment]
Hmm, probably also need to wait explocitly for config convergence, and then another minute dor maintainer to pick this up, before doing redeployment.

@bjorncs should we also let the maintainer trigger redeployment when reindexing is ready ...?

File: tests/search/schemachanges/schemachanges_need_refeed_reconfig.rb [hunk 1]
```diff
@@ -43,6 +43,14 @@ def test_need_refeed_reconfig
     assert_result("sddocname:test&nocache", @test_dir + "result.1.xml")
     assert_hitcount("f1:b&nocache", 2)
     assert_hitcount("f3:%3E29&nocache", 2)
+
+    # redeploy again to trigger reindexing, then wait for up to 5 minutes for document 1 to be reindexed
+    redeploy("test.1.sd")
+    start = Time.now
+    until /<field name="a1">30<\/field>/ =~ search("sddocname:test&nocache") or Time.now - start > 300 # seconds
```
Comment (geirst) [inline tests/search/schemachanges/schemachanges_need_refeed_reconfig.rb:50]: I don't think we should rely on regexp matching here. It is better to use the Resultset object and find the hit and field value there.

**Review by bjorncs — APPROVED** [review]

File: tests/search/schemachanges/schemachanges_need_refeed_reconfig.rb (hunk 1 above)
Comment (jonmv) [reply tests/search/schemachanges/schemachanges_need_refeed_reconfig.rb:50]: Thanks.

**Comment by jonmv** [comment]
Should also wait for container nodes, which run the indexing script, to have the new config generation. This done already, in `redeploy`, for content nodes, but not for containers.

**Comment by jonmv** [comment]
The config server itself wait for config convergence of the whole app, before setting reindexing as ready. Checking this API will suffice. 

Also found the `Resultset`, @geirst 👍

PTAL.

File: tests/search/schemachanges/schemachanges_need_refeed_reconfig.rb [hunk 2]
```diff
@@ -39,7 +39,8 @@ def test_need_refeed_reconfig
     assert_match(/Consider re-indexing document type 'test' in cluster 'search'.*\n.*Field 'f2' changed: add index aspect/, redeploy_output)
 
     # Wait up to 2 minutes for reindexing to be ready
-    until Time.now - start > 120 # seconds
+    startTime = Time.now
```
Comment (geirst) [inline tests/search/schemachanges/schemachanges_need_refeed_reconfig.rb:45]: Please use snake case for ruby code: start_time.

File: tests/search/schemachanges/schemachanges_need_refeed_reconfig.rb [hunk 3]
```diff
@@ -39,9 +39,10 @@ def test_need_refeed_reconfig
     assert_match(/Consider re-indexing document type 'test' in cluster 'search'.*\n.*Field 'f2' changed: add index aspect/, redeploy_output)
 
     # Wait up to 2 minutes for reindexing to be ready
+    reindexingUrl = "https://#{vespa.configservers["0"].name}:#{vespa.configservers["0"].ports[1]}/application/v2/tenant/#{@tenant_name}/application/#{@application_name}/environment/prod/region/default/instance/default/reindexing"
```
Comment (geirst) [inline tests/search/schemachanges/schemachanges_need_refeed_reconfig.rb:44]: Please use snake case for ruby code: reindexing_url

Comment (jonmv) [reply tests/search/schemachanges/schemachanges_need_refeed_reconfig.rb:44]: Of course!

**Comment by jonmv** [comment]
Can't get the config server API to work. Need to figure out this.

**Comment by jonmv** [comment]
PTAL.

**Comment by jonmv** [comment]
PTAL.

**Pull request merged**